### PR TITLE
Implemented web image codec instantiation for CanvasKit.

### PR DESCRIPTION
**Pull request by MichealReed**
## Description

Network image did not work when using CanvasKit, this uses an http request to retrieve the image from the given uri and provides it to skia. Since this does not use an img tag, proper CORS handling will be required.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/49725 and https://github.com/flutter/flutter/issues/55476

## Tests

Covered by existing tests.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

### CODE REVIEW

**Comment by Hixie** [comment]
This definitely needs tests.

File: lib/web_ui/test/engine/image/html_image_codec_test.dart [hunk 1]
```diff
@@ -19,7 +19,28 @@ Future<void> main() async {
...
+      await codec.getNextFrame();
+      expect(buffer.toString(), '0/100,100/100,');
+    });
+  });
+
+  group('ImageCodecUrl', () {
+    test('loads sample image from web', () async {
+      final Uri uri = Uri.base.resolve('sample_image1.png');
+      final HtmlCodec codec = await ui.webOnlyInstantiateImageCodecFromUrl(uri);
+      ;
```
Comment (yjbanov) [inline lib/web_ui/test/engine/image/html_image_codec_test.dart:34]: nit: remove

File: lib/web_ui/lib/src/engine/compositor/image.dart [hunk 2]
```diff
@@ -5,15 +5,39 @@
...
   final CkAnimatedImageCodec codec = CkAnimatedImageCodec(animatedImage);
   callback(codec);
 }
 
+/// Instantiates a [ui.Codec] backed by an `SkAnimatedImage` from Skia after requesting from URI.
+void skiaInstantiateWebImageCodec(String src, Callback<ui.Codec> callback,
+    WebOnlyImageCodecChunkCallback? chunkCallback) {
+  if (chunkCallback != null) {
+    chunkCallback!(0, 100);
+  }
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/compositor/image.dart:23]: nit: the above 3 lines can be reduced to `chunkCallback?.call(0, 100);`, I think.

Comment (MichealReed) [reply lib/web_ui/lib/src/engine/compositor/image.dart]: Fixed in `html_image_codec.dart` too.

**Comment by MOPineyro** [comment]
@MichealReed thanks for this PR, was hoping it would solve the issues, but still getting failures trying to load pngs with Network Image, any advice?

**Comment by yjbanov** [comment]
@MOPineyro please file a new issue so we can discuss it at the issue triage. It is also possible that you are hitting the CORS problem (https://github.com/flutter/flutter/issues/45955), which is on our roadmap to address.

**Pull request merged**